pkg/distributor/otlp/cmd/generate/main.go

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
This looks like it would be much simpler if it existed as a shell script.

Reply:
Actually, the drawback of making it a shell script is that you can't have the dummy dependency in the generator:

```golang
// Dummy dependency to ensure these files get vendored and we can use them as transformation sources.
_ "github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheusremotewrite"
```

Reply:
Using this for dependency is a good point, although something that can be solved easily by any other go file.

> Also, I find it much more difficult to write advanced shell script at least.

There doesn't seem to be anything advanced here, just lot of error-checking boilerplate code. Something that simple `set -v` does for you in shell script.